src/core/raster/qgsrasterlayer.cpp

```diff
@@ -2216,9 +2216,9 @@ bool QgsRasterLayer::readSymbology( const QDomNode &layer_node, QString &errorMe
     {
       const QString rendererType = rasterRendererElem.attribute( QStringLiteral( "type" ) );
       QgsRasterRendererRegistryEntry rendererEntry;
-      if ( QgsApplication::rasterRendererRegistry()->rendererData( rendererType, rendererEntry ) )
+      if ( mDataProvider && QgsApplication::rasterRendererRegistry()->rendererData( rendererType, rendererEntry ) )
```

Comment:
This isn't safe to do -- it would cause loss of style in case of a layer with a broken file path. (Same with line 2455). Is there an alternative way to fix this?

Reply:
>  Is there an alternative way to fix this?

Hum, I'm not sure. Not familiar enough with that logic to have an enlightened opinion.

Is the issue is that we now return true in readXml() without having loaded everything in FlatDontResolveLayers mode.
If returning false would solve your concern, then I've verified that doing the follow still solves #58818:

```

  if ( ! mDataProvider )
  {
    if ( !( mReadFlags & QgsMapLayer::FlagDontResolveLayers ) )
    {
      QgsDebugError( QStringLiteral( "Raster data provider could not be created for %1" ).arg( mDataSource ) );
    }

    const QDomElement layerElement = layer_node.toElement();
    readCommonStyle( layerElement, context, QgsMapLayer::AllStyleCategories );

    return false;
  }
```

At least, that's not worse than the current situation, at least if it doesn't solve entirely your concern

Reply:
Actually sorry, I'm completely wrong here. I was getting mistaken with the vector model where an invalid layer == no data provider. For rasters we still have a provider object, it's just invalid!

Sorry for the noise!